scrapper.py

```diff
@@ -26,38 +36,133 @@ class Crawler:
+    for seed_url in seed_urls:
+        if not re.match(r'https?://', seed_url):
+            raise IncorrectURLError
+    if not seed_urls:
+        raise IncorrectURLError
+    if not isinstance(seed_urls, list):
+        raise IncorrectURLError
+    if not isinstance(max_articles, int):
+        raise IncorrectNumberOfArticlesError
+    if max_articles > 100:
+        raise NumberOfArticlesOutOfRangeError
+    if max_articles == 0 or max_articles < 0:
+        raise IncorrectNumberOfArticlesError
+    prepare_environment(ASSETS_PATH)
+    return seed_urls, max_articles
+
+
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def parse(self):
+        self.article = Article(self.article_url, self.article_id)
+        headers = {'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) '
+                                 'AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36'}
+        response = requests.get(self.article_url, headers)
+        article_bs = BeautifulSoup(response.text, 'html.parser')
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        self.article.save_raw()
+        return self.article
+
+    def _fill_article_with_text(self, article_bs):
+        table_rows = article_bs.find('iframe', class_='pdf')
+        print(table_rows['data-src'])
+        url_of_pdf = table_rows['data-src']
+        pdf_raw_file = PDFRawFile(url_of_pdf, self.article_id)
+        pdf_raw_file.download()
+        self.article.text = pdf_raw_file.get_text()
+        print(len(self.article.text))
+        if 'Литература' in self.article.text:
+            new_list = self.article.text.split('Литература')
+            new_list_without_literature = new_list[:-1]
+            new_list_without_literature = ''.join(new_list_without_literature)
+            self.article.text = new_list_without_literature
+        print(len(self.article.text))
+
+    def _fill_article_with_meta_information(self, article_bs):
+        author_bs = article_bs.find('span', class_='field__item-wrapper')
+        if not author_bs:
+            author_bs = 'NOT FOUND'
+        else:
+            author_bs = author_bs.get_text()
+        title_of_the_article_bs = article_bs.find('title')
+        title_of_the_article_bs = title_of_the_article_bs.get_text()
+        data_bs = article_bs.find('div', class_='node__content clearfix')
+        data = data_bs.find('div',
+                            {'style':'text-align: left; font-weight: bold; margin-bottom: 10px;'})
+        full_data_string = data.get_text()
+        data_year_string = re.search(r'\s+\d{4}', full_data_string)
+        print(data_year_string)
+        data_year = data_year_string.group(0)[-4:]
+        date = datetime.datetime(int(data_year), 1, 1)
```

Comment:
you could be more creative. your know your journal regular schedule. from issue number you can understand what is the publication month.